chain-core/src/state/account.rs

```diff
@@ -269,7 +311,14 @@ impl StakedState {
         self.bonded = (self.bonded - bonded_slash_value)?;
         self.unbonded = (self.unbonded - unbonded_slash_value)?;
 
-        bonded_slash_value + unbonded_slash_value
+        let slash_amount = (bonded_slash_value + unbonded_slash_value)?;
+
+        if let Some(ref mut punishment) = self.punishment {
+            punishment.slash_amount = Some(slash_amount);
+            punishment.kind = punishment_kind;
+        }
```

Comment:
Similarly here, it should be enforced slash_wait_period < jail_duration, e.g. in validate_config_get_genesis